web/static/js/components/thesis-editor.js

```diff
@@ -223,6 +264,44 @@ class ThesisEditor extends React.Component {
     }
   }
 
+  importContent (t, d, page) {
+    switch (t) {
+      case 'image':
+      case 'background_image':
+        return this.imageEditor.uploadAndSet(d, page, () => { this.updateImportedContentCompletedCount() })
+      case 'text':
+        this.updateImportedContentCompletedCount()
+        return this.textEditor.set(d.name, d)
+      case 'html':
+        this.updateImportedContentCompletedCount()
+        return this.htmlEditor.set(d.name, d)
+      case 'raw_html':
+        this.updateImportedContentCompletedCount()
+        return this.rawHtmlEditor.set(d.name, d)
+      default:
+        return true
+    }
+  }
```

Comment:
we call each editor and set the content. we also increase the counter after each item and after every image finishes uploading.